ddtrace/contrib/wsgi/wsgi.py

```diff
@@ -40,6 +42,43 @@
 )
 
 
+class DDTraceWSGIIter:
+    def __init__(self, resp_iter, req_span, resp_span):
+        self._close = getattr(resp_iter, "close", None)
+        if PY2:
+            self._next = getattr(iter(resp_iter), "next")
+        else:
+            self._next = getattr(iter(resp_iter), "__next__")
+
+        self.req_span = req_span
+        self.resp_span = resp_span
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            return self._next()
+        except StopIteration:
+            self.resp_span.finish()
+            self.req_span.finish()
```

Comment:
According to [pep-333](https://peps.python.org/pep-0333/#middleware-components-that-play-both-sides) `DDTraceWSGIIter.close()` should always be called so calling StopIteration here is unnecessary but it doesn't hurt to be safe. We should always finish the spans